File: solutions/lua/darts/1/darts.lua
local Darts = {}

function Darts.score(x, y)
    local distance = math.sqrt(x * x + y * y)

    if distance > 10 then
        return 0
    elseif distance <= 1 then
        return 10
    elseif distance <= 5 then
        return 5
    else
        return 1
    end
end

return Darts

File: solutions/ruby/darts/1/darts.rb
class Darts
  def initialize(x, y)
    distance_from_bullseye = Math.sqrt(x * x + y * y)

    @score = 10
    @score = 5 if distance_from_bullseye > 1
    @score = 1 if distance_from_bullseye > 5
    @score = 0 if distance_from_bullseye > 10
  end

  attr_reader :score
end